hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/data/Data.java

```diff
@@ -92,7 +92,7 @@
 
     /** For JSON Construction ONLY */
     public Data() {
-        this(null, 0L, null);
+        this(null, System.currentTimeMillis(), null);
```

Comment:
You are right, the constructor and setTimestamp() are properly protected.
This patch is useless.
I will close it and investigate further what could be the root cause.